chore: Adds guide on reasons and steps for deleting a BCP-enabled cluster (#3409)
mongodb/terraform-provider-mongodbatlas: docs/data-sources/advanced_cluster (preview provider 2.0.0).md

<br> &#8226; Changes to cluster configurations can affect costs. Before making changes, please see [Billing](https://docs.atlas.mongodb.com/billing/).
<br> &#8226; If your Atlas project contains a custom role that uses actions introduced in a specific MongoDB version, you cannot create a cluster with a MongoDB version less than that version unless you delete the custom role.

> -> **NOTE:** If you want to delete an Atlas Cluster that has an associated `mongodbatlas_cloud_backup_schedule` resource and a Backup Compliance Policy enabled, read [this guide](../guides/delete-cluster-with-backup-compliance-policy.md).

Reviewer: Nit: instead of "read this guide" I would say here and in all such places: see [Backup Compliance Policy](link)

The goal is to use the same formula: To learn more, see link. And to use the actual title or close to the actual title in the link's title, not generic words like "here", or "this guide". Hope this helps! Not a stopper though, but good practice.

Author: Hi @JuliaMongo, I understand you - I am just not convinced that "Backup Compliance Policy" is a good title representing the guide. Arguably, "this guide" is not either.

so I wonder: is "see [Delete cluster with Backup Compliance Policy](link)" too long? or ok?

Reviewer: Ah, I see. How about:

```
To delete an Atlas cluster that has an associated `mongodbatlas_cloud_backup_schedule` 
resource and an enabled Backup Compliance Policy, first instruct Terraform to ignore
the `mongodbatlas_cloud_backup_schedule` resource and then use Terraform to
delete the cluster. To learn more, see 
[Delete a Cluster with a Backup Compliance Policy](../guides/delete-cluster-with-backup-compliance-policy.md).

```

> -> **NOTE:** If you want to delete an Atlas Cluster that has an associated `mongodbatlas_cloud_backup_schedule` resource and a Backup Compliance Policy enabled, read the [following guide](../guides/delete-cluster-with-backup-compliance-policy.md).

Reviewer 2: ```suggestion
-> **NOTE:** If you want to delete an Atlas Cluster that has an associated `mongodbatlas_cloud_backup_schedule` resource and a Backup Compliance Policy enabled, read [this guide](../guides/delete-cluster-with-backup-compliance-policy.md).
```
